tests/test_cli.py

```diff
@@ -0,0 +1,114 @@
+"""
+Tests the cli
+"""
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import unittest
+import argparse
+
+import tests
+import ga4gh.cli as cli
+import ga4gh.protocol as protocol
+import ga4gh.client as client
+
+
+class TestNoInput(unittest.TestCase):
+    """
+    Test the cli as if there was no input; print_help should be called
+    """
+    def testServerNoInput(self):
+        parser = StubArgumentParser()
+        cli.server_main(parser)
+        parser.assertCounts(1, 1)
+
+    def testCliNoInput(self):
+        parser = StubArgumentParser()
+        cli.client_main(parser)
+        parser.assertCounts(1, 1)
+
+
+class TestClientArguments(unittest.TestCase):
+    """
+    Tests the client cli can parse all arguments it is supposed to
+    and can initialize the runner in preparation for a request
+    """
+    def setUp(self):
+        # initialize the client parser
+        self.parser = StubArgumentParser()
+        cli.client_main(self.parser)
+
+    def tearDown(self):
+        # include arguments common to all commands
+        inputStr = "--verbose --workarounds WORK,AROUND --key KEY {0} URL"
+        cliInput = inputStr.format(self.cliInput)
+        splits = cliInput.split()
+
+        # parse the arguments
+        args = self.parser.parser.parse_args(splits)
+
+        # invoke the initializer of the runner (which also parses args)
+        runner = args.runner(args)
+
+        # ensure the correct attributes on the runner are set
+        assert isinstance(runner._request, protocol.ProtocolElement)
```

Comment:
The unittest docs say that avoiding `assert` is better, so we should do that.  I'll eliminate existing `asserts` from the tests.